### PR TITLE
Make build version number dynamic in useragent string

### PR DESCRIPTION
**Pull request by sdodsley**

### CODE REVIEW

File: build/docker/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,7 @@ COPY go.mod go.sum ./
 RUN go mod download && go mod verify
 
 COPY . .
-RUN CGO_ENABLED=1 go build -mod=readonly -a -tags 'netgo osusergo static_build' -ldflags="-X main.version=v$VERSION" -v -o /usr/local/bin/pure-fb-om-exporter cmd/fb-om-exporter/main.go
+RUN CGO_ENABLED=1 go build -mod=readonly -a -tags 'netgo osusergo static_build' -ldflags="-X 'main.version=v$VERSION" -X 'purestorage/fb-openmetrics-exporter/internal/rest-client.UserAgentString=$(VERSION)'" -v -o /usr/local/bin/pure-fb-om-exporter cmd/fb-om-exporter/main.go
```
Comment (chrroberts-pure) [inline build/docker/Dockerfile:11]:
```suggestion
RUN CGO_ENABLED=1 go build -mod=readonly -a -tags 'netgo osusergo static_build' -ldflags="-X 'main.version=v$VERSION' -X 'purestorage/fb-openmetrics-exporter/internal/rest-client.UserAgentString=$(VERSION)'" -v -o /usr/local/bin/pure-fb-om-exporter cmd/fb-om-exporter/main.go
```

**Comment by chrroberts-pure** [comment]
Looks good. Review the unbalanced `'` and I’ll test, and also probably throw in a `make update-go-deps` in another PR to update`resty`

File: build/docker/Dockerfile (hunk 1 above)
Comment (sdodsley) [reply build/docker/Dockerfile]: @chrroberts-pure typo fixed

**Review by chrroberts-pure — CHANGES_REQUESTED** [review]
Looks good after these changes.

**Review by chrroberts-pure — APPROVED** [review]
LGTM - Resty update coming in soon

**Pull request merged**